elfi/visualization/visualization.py

```diff
@@ -271,9 +271,13 @@ def progress_bar(iteration, total, prefix='', suffix='', decimals=1, length=100,
         Bar fill character
 
     """
-    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
-    filled_length = int(length * iteration // total)
-    bar = fill * filled_length + '-' * (length - filled_length)
-    print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end='\r')
-    if iteration == total:
-        print()
+    try:
+        percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+        filled_length = int(length * iteration // total)
+        bar = fill * filled_length + '-' * (length - filled_length)
+        print('\r%s |%s| %s%% %s' % (prefix, bar, percent, suffix), end='\r')
+        if iteration == total:
+            print()
+    except ZeroDivisionError:
+        print("Can't output progress bar. Number of total iterations is zero. \n"
+              "Please check your model description or put bar argument to False.")
```

Comment:
I think `try` is overkill here. Just put the whole thing under `if total > 0`, no else block necessary I suppose.